Mayuko-branch

kondomayuko opened the pull request.

Commit:
vault1 and vault2 contracts and all tests passed

shubham-kanodia commented on contracts/Vault2.sol:
```diff
@@ -1,6 +1,28 @@
+        require(_amount!=0, "amount cannot be zero");
+        require(msg.value == _amount, "message value and amount must match");
+        _mint(msg.sender, _amount);
+        emit Minted(msg.sender, _amount);
```
_mint already emits an event indicating a mint - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L267

shubham-kanodia commented on contracts/Vault2.sol:
```diff
@@ -1,6 +1,28 @@
+        _burn(msg.sender, _amount);
+        (bool success, ) = payable(msg.sender).call{value: _amount}("");
+        require(success, "Failed to send Ether");
+        emit Burned(msg.sender, _amount);
```
_burn also emits an event indicating a burn - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L295

kondomayuko committed:
Update Vault2.sol

eliminated "mint" and "burned" events because it is already defined in openzepplin ERC20 file.

kondomayuko committed:
Update Vault2Test.js

eliminated event tests.

konmaru-fox commented:
Thank you very much for your review! I eliminated "mint" and "burn" events that are already defined in openzepplin ERC20 file.
